dozer/cogs/filter.py

```diff
@@ -72,17 +72,57 @@ async def check_filters(self, message):
                     await message.delete()
                     deleted = True
 
+    async def check_filters_nicknames(self, member_before, member_after):
+        """Check all filters for a members nickname change"""
+        if member_after.id == self.bot.user.id or not hasattr(member_after, 'roles'):
+            return
+        roles = await self.word_filter_role_whitelist.query_all(guild_id=member_after.guild.id)
+        whitelisted_ids = set(role.role_id for role in roles)
+        if any(x.id in whitelisted_ids for x in member_after.roles):
+            return
+        try:
+            filters = self.filter_dict[member_after.guild.id]
+        except KeyError:
+            await self.load_filters(member_after.guild.id)
+            filters = self.filter_dict[member_after.guild.id]
+        reverted = False
+        if member_after.nick is None:
+            return
+        for wordid, wordfilter in filters.items():
+            if wordfilter.search(member_after.nick) is not None:
+                time = datetime.datetime.utcnow()
+                infraction = WordFilterInfraction(member_id=member_after.id, filter_id=wordid,
+                                                  timestamp=time,
+                                                  message=member_after.nick)
+                await infraction.update_or_add()
+                if not reverted:
+                    try:
+                        await member_after.edit(nick=member_before.nick)
+                        await member_after.send(f"{member_after.mention}, your nickname in **{member_after.guild}** "
+                                                f"contained a banned word and has been reset to your previous nickname")
+                    except discord.Forbidden:
+                        await member_after.send(f"{member_after.mention}, your nickname in **{member_after.guild}** "
```

Comment:
Perhaps this should also be reported to mods in some way? This might be a better fit for the filter logging stuff I have planned.